src/gui/qgsmaskingwidget.cpp

```diff
@@ -107,15 +87,15 @@ QList<QPair<QString, QList<QgsSymbolLayerReference>>> symbolLayerMasks( const Qg
 
 void QgsMaskingWidget::setLayer( QgsVectorLayer *layer )
 {
-  if ( mLayer != layer )
-  {
-    mLayer = layer;
-    mMustPopulate = true;
-  }
+  mLayer = layer;
+  populate();
```

Comment:
isn't it still worth avoiding the function call if mLayer == layer ?

Reply:
No because it is used as a way to refresh the widget even if the layer is the same

Reply:
Make sense. Sometimes layer mask themselves. Think of a contour line layer with labels. The labels are in the same layer than the lines that need to be masked.

Reply:
> No because it is used as a way to refresh the widget even if the layer is the same

should there be an explicit "refreshWidget" then, maybe ?

Reply:
If I understand correctly what you propose, That would involve calling code to test whether call setLayer if layer is different from current one or call refreshWidget if it's the same, so more code to maintain. I don't see a added value of such method